src/components/CocoaPodsVersionPicker.js

```diff
@@ -0,0 +1,95 @@
+// (c) Copyright 2022, SAP SE and ClearlyDefined contributors. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { getCocoaPodsRevisions } from '../api/clearlyDefined'
+import Autocomplete from './Navigation/Ui/Autocomplete'
+import searchSvg from '../images/icons/searchSvg.svg'
+
+export default class CocoaPodsVersionPicker extends Component {
+  static propTypes = {
+    onChange: PropTypes.func,
+    request: PropTypes.object.isRequired,
+    defaultInputValue: PropTypes.string
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      customValues: [],
+      options: [],
+      focus: false,
+      selected: props.request.revision ? [props.request.revision] : []
+    }
+    this.onChange = this.onChange.bind(this)
+    this.filter = this.filter.bind(this)
+  }
+
+  componentDidMount() {
+    this.getOptions('')
+  }
+
+  componentWillReceiveProps(nextProps) {
```

Comment:
Setting "selected" in componentDidUpdate has an undesirable side effect: after typing a valid version in the version picker, backspace once (should only clear one letter) triggered clearing all the letters user typed. Change to calculating "selected" at rendering time based on this.props.

Reply:
the undesirable side effect is happening because this.state.selected continues to rely on props.request.revision after initializing the component (as shown on line 34).

Why not just determine what to pass through at the parent and pass selected directly to the Autocomplete component as a prop instead?

Reply:
Thanks for the suggestion.  I am new to react and really appreciate your input!  The parent (HarvestQueueList) maintains a list of harvest queue requests. HarvestQueueList understands request and request.revision, but it does not need to know the implementation detail of how request.revision is displayed by its children (in a text box or an AutoComplete). The version display logic is encapsulated in all the version pickers (including CocoaPodVersionPicker), which convert the request.revision to "selected" for AutoComplete.  This encapsulation is beneficial when a different version picker is to be used in the future, only the version picker needs to be changed, and no change is necessary for the parent, HarvestQueueList.  So setting "selected" in CocoaPodVersionPicker makes more sense to me.

In CocoaPodVersionPicker, the conversion from request.revision to "selected" is inexpensive, and can be done at rendering time without being kept as a separate state.  See https://github.com/clearlydefined/website/pull/971/commits/b42eb4e469e9bdec608f5b28a151aef6b5e8315d